lib/views/pay/numberpad_view.dart

```diff
@@ -93,143 +93,49 @@ class _NumberPadState extends State<NumberPad> {
       body: SafeArea(
         bottom: true,
         top: true,
-        child: Column(
-          children: [
-            const Spacer(flex: 1),
-            Expanded(
-              flex: 1,
-              child: Text(
-                display,
-                textScaleFactor: 1.5,
-                style: const TextStyle(
-                  fontSize: 40,
-                  fontWeight: FontWeight.bold,
-                ),
-              ),
-            ),
-            const Spacer(flex: 1),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '1'),
-                  numberButton(size, '2'),
-                  numberButton(size, '3')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '4'),
-                  numberButton(size, '5'),
-                  numberButton(size, '6')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '7'),
-                  numberButton(size, '8'),
-                  numberButton(size, '9')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '.'),
-                  numberButton(size, '0'),
-                  numberButton(
-                    size,
-                    '',
-                    icon: Icons.backspace_outlined,
-                  ),
-                ],
-              ),
-            ),
-            const Spacer(flex: 1),
-            Expanded(
-              child: MainCircleButton(
-                  icon: Icons.send_outlined,
-                  label: "Pay",
-                  onPress: () async {
-                    if (widget.invoice != null) {
-                      payInvoice(widget.invoice!, int.parse(display));
-                    } else {
-                      withdraw(widget.btcAddress!, int.parse(display));
-                    }
-                  }),
-            ),
-          ],
-        ),
-      ),
-    );
-  }
+        child: NumberPadBody(
+          size: MediaQuery.of(context).size,
+          display: display,
+          onNumberButtonTap: (value, icon) {
+            if (icon != null) {
+              if (display.length == 1) {
+                display = '0';
+              } else {
+                display = display.substring(0, display.length - 1);
+              }
```

Comment:
Why do not put the logic that we had as default inside the trash library? And if the user wants can override it by providing a `onNumberButtonTap` callback?